pkg/controllers/dynakube/oneagent/daemonset/arguments.go

```diff
@@ -26,7 +26,7 @@ func (b *builder) arguments() ([]string, error) {
 		return []string{}, err
 	}
 
-	if !isProxyAsEnvDeprecated {
+	if !isProxyAsEnvDeprecated || b.dk.FeatureOneAgentIgnoreProxy() {
```

Comment:
```suggestion
	if !isProxyAsEnvDeprecated || b.dk.NeedsOneAgentProxy() {
```
use the more broad check

Reply:
problem that this check is incorrect in this situation, I've tried:
```
func (dk *DynaKube) NeedsOneAgentProxy() bool {
	return !dk.FeatureOneAgentIgnoreProxy() && dk.HasProxy()
}
```
and then switched back to this one.

Reply:
I tried your branch, and the bug is still there, because the feature-flag is only a way to make the bug happen

What I did:
1. Deploy OA with proxy
2. Check contents of `data/zib50933/osagent/var/agent/config/ruxitagentproc.conf` (so the OA config)
```
cat data/zib50933/osagent/var/agent/config/ruxitagentproc.conf | grep proxy
proxy :@squid.proxy.svc.cluster.local:3128
```
4. Undeploy OA
5. Deploy OA without proxy (didn't use the feature-flag)
6. (again )Check contents of `data/zib50933/osagent/var/agent/config/ruxitagentproc.conf` (so the OA config)
```
cat data/zib50933/osagent/var/agent/config/ruxitagentproc.conf | grep proxy
proxy :@squid.proxy.svc.cluster.local:3128
```

Reply:
Edit: I updated by suggestion to `!b.dk.NeedsOneAgentProxy()` (my mistake)

Reply:
yeah I thought about this, but i'm not sure. Have you tried with this fix? 
```
!b.dk.NeedsOneAgentProxy()
```
if we rewrite it:
```
dk.FeatureOneAgentIgnoreProxy() && !dk.HasProxy()
```

Reply:
okey let's do it.

Reply:
> Have you tried with this fix?
I updated the unittests, and it works

> `dk.FeatureOneAgentIgnoreProxy() && !dk.HasProxy()`

```
func (dk *DynaKube) NeedsOneAgentProxy() bool {
	return !dk.FeatureOneAgentIgnoreProxy() && dk.HasProxy()
}
```
so `!dk.NeedsOneAgentProxy()` == `dk.FeatureOneAgentIgnoreProxy() || !dk.HasProxy()` (and this is what we need)

Reply:
yeah on it 
<img src="https://media0.giphy.com/media/qadvd1vBaZBBu/giphy.gif"/>

Reply:
I have the tests updated, can just push it if you want :D